tests/std/tests/Dev11_0535636_functional_overhaul/test.cpp

```diff
@@ -2169,6 +2159,42 @@ void test_more_bind() {
     STATIC_ASSERT(
         is_same_v<decltype(p11), Pack<const Typewriter&, const bool&,
                                      Pack<const Typewriter&, Z&, const Z&, Z&&, const Z&&>&&, Z&, const double&>>);
+
+#if _HAS_CXX17
+    struct NothrowInvocable {
+        int operator()(int i) noexcept {
+            return i;
+        }
+    };
+    struct NotNothrowInvocable {
+        int operator()(int i) {
+            return i;
+        }
+    };
+    struct PossiblyThrowingInt {
+        int i;
+        PossiblyThrowingInt() = default;
+        PossiblyThrowingInt(int j) : i(j) {}
+        operator int() {
+            return i;
+        }
+    };
+
+    static_assert(is_nothrow_invocable_v<decltype(bind(NothrowInvocable{}, 0))>);
+    static_assert(is_nothrow_invocable_v<decltype(bind(NothrowInvocable{}, _1)), int>);

```

Comment:
This is great - thanks for the test coverage! I especially like the `PossiblyThrowingInt` for both argument and return conversions. I think this should be further extended to cover all of the "arg fixup" codepaths that you carefully updated - I'll try to add that.

Reply:
Added and the tests are passing - thanks for getting to the other suggestions!